src/Tools.js

```diff
@@ -60,5 +60,33 @@ class Tools {
 		}
 		return $result;
 	}
+
+	/**
+	 * Use jQuery objects (or strings) as i18n message parameters,
+	 * without having to parse translator-supplied strings as HTML.
+	 *
+	 * @param {string} msg The i18n message name.
+	 * @param {jQuery|string} params jQuery objects or strings to insert into the message.
+	 * @return {string}
+	 */
+	static i18nHtml( msg, ...params ) {
+		const
+			// Temporary marker string that will be replaced (with an integer appended).
+			// Must not appear in messages.
+			marker = '@@I18N',
+			// Create a list of parameter markers.
+			replacements = Object.keys( params ).map( function ( k ) { return marker + k; } );
```

Comment:
I wasn't going to make this a blocker, but since I have another comment below, I'm going to channel @MusikAnimal and recommend you ES6'ify this 

`.map( k => marker + k );` should work, I think.